File: Packages/Flip54Core/Sources/Flip54Core/Rank.swift
public enum Rank: String, Codable, CaseIterable, Hashable, Sendable {
    case two, three, four, five, six, seven, eight, nine, ten
    case jack, queen, king, ace

    /// Numeric value for pip-as-reps. Returns nil for face cards and Aces.
    public var pipValue: Int? {
        switch self {
        case .two:   return 2
        case .three: return 3
        case .four:  return 4
        case .five:  return 5
        case .six:   return 6
        case .seven: return 7
        case .eight: return 8
        case .nine:  return 9
        case .ten:   return 10
        case .jack, .queen, .king, .ace: return nil
        }
    }

    public var isFace: Bool {
        self == .jack || self == .queen || self == .king
    }

    public var displaySymbol: String {
        switch self {
        case .two:   return "2"
        case .three: return "3"
        case .four:  return "4"
        case .five:  return "5"
        case .six:   return "6"
        case .seven: return "7"
        case .eight: return "8"
        case .nine:  return "9"
        case .ten:   return "10"
        case .jack:  return "J"
        case .queen: return "Q"
        case .king:  return "K"
        case .ace:   return "A"
        }
    }
}

File: android/core/src/main/kotlin/com/flip54/core/Rank.kt
package com.flip54.core

import kotlinx.serialization.Serializable

@Serializable
enum class Rank {
    TWO, THREE, FOUR, FIVE, SIX, SEVEN, EIGHT, NINE, TEN,
    JACK, QUEEN, KING, ACE;

    val pipValue: Int? get() = when (this) {
        TWO   -> 2
        THREE -> 3
        FOUR  -> 4
        FIVE  -> 5
        SIX   -> 6
        SEVEN -> 7
        EIGHT -> 8
        NINE  -> 9
        TEN   -> 10
        JACK, QUEEN, KING, ACE -> null
    }

    val isFace: Boolean get() = this == JACK || this == QUEEN || this == KING

    val displaySymbol: String get() = when (this) {
        TWO   -> "2"
        THREE -> "3"
        FOUR  -> "4"
        FIVE  -> "5"
        SIX   -> "6"
        SEVEN -> "7"
        EIGHT -> "8"
        NINE  -> "9"
        TEN   -> "10"
        JACK  -> "J"
        QUEEN -> "Q"
        KING  -> "K"
        ACE   -> "A"
    }
}
